dotCMS/src/main/java/com/dotmarketing/common/reindex/ReindexThread.java

```diff
@@ -116,15 +117,15 @@ private enum ThreadState {
             "WAIT_BEFORE_PAUSE_SECONDS", 0);
 
 
-    private AtomicReference<ThreadState> state = new AtomicReference<>(ThreadState.STOPPED);
+    private final AtomicReference<ThreadState> state = new AtomicReference<>(ThreadState.STOPPED);
 
 
-    private final static String REINDEX_THREAD_PAUSED = "REINDEX_THREAD_PAUSED";
-    private final static Lazy<SystemCache> cache = Lazy.of(() -> CacheLocator.getSystemCache());
+    private static final String REINDEX_THREAD_PAUSED = "REINDEX_THREAD_PAUSED";
+    private static final Lazy<SystemCache> cache = Lazy.of(CacheLocator::getSystemCache);
```

Comment:
cachelocator class has a lot of static and might not be loaded here.  This could potentially force its loading out of order?  Not sure but seems safer to stick with the `()->`notation?

Reply:
The case we had before where we could not convert was a very particular and  case related to a method being being passed in as the argument of the method reference,  and not just the fact it has statics.   I will do a quick trace to confirm If the initialization of the CacheLocator class could be modified due to this, but I don't think it will change the order of operations in a negative way here.    I don't want us to be afraid of using method references because we do not understand their impact because of one exception to the rule.

Reply:
I'm all about method references where they make sense and 9 times out of 10 they do. I am just scared by our old code and all their static initializers.